pkg/messaging/nats/pubsub.go

```diff
@@ -77,7 +78,14 @@ func (ps *pubsub) Subscribe(id, topic string, handler messaging.MessageHandler)
 	case true:
 		// Check topic ID
 		if _, ok := s[id]; ok {
-			return ErrAlreadySubscribed
+			// Unlocking, so that Unsubscribe() can access ps.subscriptions
+			ps.mu.Unlock()
+			if err := ps.Unsubscribe(id, topic); err != nil {
+				return err
+			}
+			ps.mu.Lock()
+			s = make(map[string]subscription)
```

Comment:
You need to replace only an entry in the `s[i]`, not the entire `ps.subscriptions[topic]`. That is, for the given topic (`ps.subscriptions[topic]`), you need to replace an existing subscription under the given ID (`ps.subscriptions[topic][id]`)